zone-setup/src/bin/zone-setup.rs

```diff
@@ -97,7 +97,7 @@ struct ChronySetupArgs {
     #[arg(
         short,
         long,
-        num_args = 1..,
+        num_args = 0..,
         value_parser = NonEmptyStringValueParser::default(),
     )]
     servers: Vec<String>,
```

Comment:
If I'm understanding correctly:

- for boundary servers, this argument is essentially required (there must be at least one value in the `Vec`)
- for internal NTP servers, this argument is not used (and should not be provided)

Is that right?  (Might be worth adding a comment. (edit: I found it in the SMF manifest.  Maybe worth mentioning here?))

Reply:
That's mostly right, but it's slightly uglier than that. The SMF manifest for `chrony-setup` unconditionally passes the `-s` flag followed by any configured servers: https://github.com/oxidecomputer/omicron/blob/356c9bf624aba3faf0b9b35239bfef4704f743c3/smf/chrony-setup/manifest.xml#L15

The change from 1 to 0 here allows that manifest to pass `... -s -a ...`; i.e., the flag with no arguments to it. We should probably clean that up - I can file an issue? I assume that would involve replacing the exec command with a bash script a la our various [method_script.sh](https://github.com/oxidecomputer/omicron/blob/356c9bf624aba3faf0b9b35239bfef4704f743c3/smf/cockroachdb/method_script.sh) helpers so we could do some checking on which flags to pass?

Reply:
I ported over the comments (and added a check that `clap` was previously doing for us, oops) in 48ee15bad. Still leaning toward filing an issue to make a cleanup pass over the way this binary accepts args from SMF, unless you think that's not worth the trouble.

Reply:
> Still leaning toward filing an issue to make a cleanup pass over the way this binary accepts args from SMF, unless you think that's not worth the trouble.

I've no opinion on that.